src/main/java/org/testng/internal/ClassImpl.java

```diff
@@ -128,44 +132,22 @@ private Object getInstanceFromGuice() {
     return injector.getInstance(m_class);
   }
 
-  @SuppressWarnings("unchecked")
   public Injector getParentInjector() {
     ISuite suite = m_testContext.getSuite();
     // Reuse the previous parent injector, if any
     Injector injector = suite.getParentInjector();
     if (injector == null) {
-      String stageString = suite.getGuiceStage();
-      Stage stage;
-      if (isStringNotEmpty(stageString)) {
-        stage = Stage.valueOf(stageString);
-      } else {
-        stage = Stage.DEVELOPMENT;
-      }
-      if (m_hasParentModule) {
-        Class<Module> parentModule = (Class<Module>) ClassHelper.forName(suite.getParentModule());
-        if (parentModule == null) {
-          throw new TestNGException(
-              "Cannot load parent Guice module class: " + suite.getParentModule());
-        }
-        Module module = newModule(parentModule);
-        injector = com.google.inject.Guice.createInjector(stage, module);
-      } else {
-        injector = com.google.inject.Guice.createInjector(stage);
+      Module parentModule = GuiceHelper.getParentModule(m_testContext);
```

Comment:
What do you think if we add the `parentModule` logic directly into `createInjector` method?

Reply:
this suggestion makes `ClassImpl` a lot nicer, but the `GuiceHelper.getInjector` a little weirder because of the injector caching.  I think its ok though?